registry/session.go

```diff
@@ -281,6 +281,9 @@ func (r *Session) GetRepositoryData(remote string) (*RepositoryData, error) {
 	// TODO: Right now we're ignoring checksums in the response body.
 	// In the future, we need to use them to check image validity.
 	if res.StatusCode != 200 {
+		if res.StatusCode == 404 {
```

Comment:
I agree with both remarks. @dmcgowan let us know if you can update, or we'll merge this as is for 1.6.0 RC1 and review later.